src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -89,7 +62,9 @@ export const ButtonGroup: ChakraComponent<
           sx={styles}
           {...rest}
         >
-          {newChildren}
+          <ButtonGroupContext.Provider value={isDisabled}>
```

Comment:
~I like the `useButtonGroup` hook pattern more than this approach, but nbd.~
Nvm, got the file mixed up, that hook is used in `Button.tsx`.